src/usdb_syncer/json.py

```diff
@@ -0,0 +1,79 @@
+"""Generates a JSON from the passed song list."""
+
+import datetime
+import json
+from typing import Any, Iterable
+
+from usdb_syncer import SongId
+from usdb_syncer.constants import VERSION
+from usdb_syncer.logger import get_logger
+from usdb_syncer.resource_dl import _url_from_resource
+from usdb_syncer.song_txt.headers import Headers
+from usdb_syncer.usdb_scraper import UsdbSong
+
+_logger = get_logger(__file__)
+
+MAX_SIZE = (100, 100)
+
+
+def get_headers(txt_path: str) -> Headers:
+    with open(txt_path, "r", encoding="utf-8") as file:
+        lines = [line for line in file.read().splitlines() if line]
+        return Headers.parse(lines, _logger)
+
+
+def generate_song_json(songs: Iterable[SongId], path: str) -> int:
+    date = datetime.datetime.now()
+    song_list: list[dict[str, Any]] = [
+        {
+            "id": song.song_id,
+            "artist": headers.artist,
+            "title": headers.title,
+            "year": int(headers.year) if headers.year else None,
+            "edition": (
+                None
+                if (not headers.edition or headers.edition == "None")
+                else headers.edition
+            ),
+            "genre": (
+                None
+                if (not headers.genre or headers.genre == "None")
+                else headers.genre
+            ),
+            "language": song.language,
+            "golden_notes": bool(song.golden_notes),  # force true/false instead of 1/0
```

Comment:
fixed in 6596382, though unsatisfying to just put the `bool(row[5])` there, I hoped to find some db table column definition where I would expect golden_notes to be defined as `INT` that I could turn into `BOOLEAN` so that database already returns a proper data type, but couldn't find it.

Reply:
SQLite only supports data types TEXT, NUMERIC, INTEGER, REAL and BLOB.